pgjdbc/src/test/java/org/postgresql/test/jdbc42/TimestampUtilsTest.java

```diff
@@ -62,6 +71,44 @@ public void testToLocalTime() throws SQLException {
     assertEquals(LocalTime.parse("23:59:59.999999999"), timestampUtils.toLocalTime("24:00:00"));
   }
 
+  @Test
+  public void testLocalDateTimeRounding() {
+
+    assertLocalDateTimeRounding("2022-01-04T08:57:13.123457", "2022-01-04T08:57:13.123456500");
+    assertLocalDateTimeRounding("2022-01-04T08:57:13.123456", "2022-01-04T08:57:13.123456499");
+    assertLocalDateTimeRounding("2022-01-04T08:57:14","2022-01-04T08:57:13.999999501");
+
+    LocalDateTime unrounded = LocalDateTime.parse("2022-01-04T08:57:13.123456");
+    assertSame(localDateTimeRoundedFailureMessage(unrounded), unrounded, timestampUtils.round(unrounded));
```

Comment:
@marschall , I know I might be picky to the assertion messages, however, I would suggest doing something different here.

`unrounded` would be printed in case of failure anyway, so `localDateTimeRoundedFailureMessage(unrounded)` adds no value.

I would suggest add **the reason** you believe the thing should come out "the same".


```suggestion
    assertSame(
        "nanosecond part is 000, so timestampUtils.round should not modify the value, and keep the same object as result for performance reasons",
        unrounded,
        timestampUtils.round(unrounded)
    );
```